troupon/staticfiles/admin/js/inlines.js

```diff
@@ -0,0 +1,272 @@
+          for (var i=0, formCount=forms.length; i<formCount; i++)
+          {
+            updateElementIndex($(forms).get(i), options.prefix, i);
+            $(forms.get(i)).find("*").each(function() {
+              updateElementIndex(this, options.prefix, i);
+            });
+          }
+        });
+        // If a post-add callback was supplied, call it with the added form:
+        if (options.added) {
+          options.added(row);
+        }
+      });
+    }
+    return this;
+  };
+
+  /* Setup plugin defaults */
+  $.fn.formset.defaults = {
+    prefix: "form",          // The form prefix for your django formset
+    addText: "add another",      // Text for the add link
+    deleteText: "remove",      // Text for the delete link
+    addCssClass: "add-row",      // CSS class applied to the add link
+    deleteCssClass: "delete-row",  // CSS class applied to the delete link
+    emptyCssClass: "empty-row",    // CSS class applied to the empty row
+    formCssClass: "dynamic-form",  // CSS class applied to each form in a formset
+    added: null,          // Function called each time a new form is added
+    removed: null          // Function called each time a form is deleted
+  };
+
+
+  // Tabular inlines ---------------------------------------------------------
+  $.fn.tabularFormset = function(options) {
+    var $rows = $(this);
+    var alternatingRows = function(row) {
+      $($rows.selector).not(".add-row").removeClass("row1 row2")
+        .filter(":even").addClass("row1").end()
+        .filter(":odd").addClass("row2");
+    };
+
+    var reinitDateTimeShortCuts = function() {
+      // Reinitialize the calendar and clock widgets by force
+      if (typeof DateTimeShortcuts != "undefined") {
+        $(".datetimeshortcuts").remove();
+        DateTimeShortcuts.init();
+      }
+    };
+
+    var updateSelectFilter = function() {
+      // If any SelectFilter widgets are a part of the new form,
+      // instantiate a new SelectFilter instance for it.
+      if (typeof SelectFilter != 'undefined'){
+        $('.selectfilter').each(function(index, value){
+          var namearr = value.name.split('-');
+          SelectFilter.init(value.id, namearr[namearr.length-1], false, options.adminStaticPrefix );
+        });
+        $('.selectfilterstacked').each(function(index, value){
+          var namearr = value.name.split('-');
+          SelectFilter.init(value.id, namearr[namearr.length-1], true, options.adminStaticPrefix );
+        });
+      }
+    };
+
+    var initPrepopulatedFields = function(row) {
+      row.find('.prepopulated_field').each(function() {
+        var field = $(this),
+            input = field.find('input, select, textarea'),
+            dependency_list = input.data('dependency_list') || [],
+            dependencies = [];
+        $.each(dependency_list, function(i, field_name) {
+          dependencies.push('#' + row.find('.field-' + field_name).find('input, select, textarea').attr('id'));
```

Comment:
Line is too long.<br>Mixed double and single quotes.<br>Identifier 'field_name' is not in camel case.